ci-operator/step-registry/api-job/k8sgpt/api-job-k8sgpt-commands.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+
+set -e
+set -u
+set -o pipefail
+
+openai_token=$(cat "/var/run/vault/tests-private-account/openai-token")
+
+k8sgpt version
+set +x
+k8sgpt auth add -m gpt-3.5-turbo -b openai -p ${openai_token} | tee ${ARTIFACT_DIR}/k8sgpt-result || true
+set -x
+
+k8sgpt analyze -aed | tee -a ${ARTIFACT_DIR}/k8sgpt-result || true
+
+log=`cat ${ARTIFACT_DIR}/k8sgpt-result`
+
+if [[ ${log} =~ "No problems detected" ]]; then
```

Comment:
@jianzhangbjz I just find the junit schema is not correct. it should be the follwoing
```shell
if [[ ${log} =~ "No problems detected" ]]; then
    cat >"${ARTIFACT_DIR}/k8sgpt-result.xml" <<EOF
    <testsuite name="api-job-k8sgpt" tests="1" failures="0">
        <testcase name="scanning cluster"/>
    </testsuite>
EOF
else
    cat >"${ARTIFACT_DIR}/k8sgpt-result.xml" <<EOF
    <testsuite name="api-job-k8sgpt" tests="1" failures="1">
      <testcase name="scanning cluster">
        <failure message="problems are detected">
        </failure>
        <system-out>
          ${log}
        </system-out>
      </testcase>
    </testsuite>
EOF
fi
```

Reply:
@jianzhangbjz I am sorry I made mistake.
it should be 
```shell
if [[ ${log} =~ "No problems detected" ]]; then
    cat >"${ARTIFACT_DIR}/k8sgpt-result.xml" <<EOF
    <testsuite name="api-job-k8sgpt" tests="1" failures="0">
        <testcase name="scanning cluster"/>
    </testsuite>
EOF
else
    cat >"${ARTIFACT_DIR}/k8sgpt-result.xml" <<EOF
    <testsuite name="api-job-k8sgpt" tests="1" failures="1">
      <testcase name="scanning cluster">
        <failure message="">fail problems are detected</failure>
        <system-out>
          ${log}
        </system-out>
      </testcase>
    </testsuite>
EOF
fi
```

the failure message format I ever made is wrong.

current is correct which should be analysis by reportportal.